public/app/features/dashboard/components/Inspector/PanelInspector.tsx

```diff
@@ -110,8 +108,24 @@ export class PanelInspector extends PureComponent<Props, State> {
 
     // Find the first DataSource wanting to show custom metadata
     let metaDS: DataSourceApi;
-    const data = lastResult?.series;
-    const error = lastResult?.error;
+    const data = lastResult.series;
+    const error = lastResult.error;
+
+    const targets = lastResult.request?.targets;
+    const requestTime = lastResult.request?.endTime ? lastResult.request?.endTime - lastResult.request.startTime : -1;
+    const queries = targets ? targets.length : 0;
+    const temp: { [key: string]: { count: number } } = {};
```

Comment:
this datasources count can be replaced with `new Set(targets.map(t => t.datasource)).size` :)